packages/flutter_tools/lib/src/macos/build_macos.dart

```diff
@@ -17,21 +17,22 @@ import '../project.dart';
 // TODO(jonahwilliams): support target option.
 // TODO(jonahwilliams): refactor to share code with the existing iOS code.
 Future<void> buildMacOS(FlutterProject flutterProject, BuildInfo buildInfo) async {
+  final Directory flutterBuildDir = fs.directory(getMacOSBuildDirectory());
+  if (!flutterBuildDir.existsSync()) {
+    flutterBuildDir.createSync(recursive: true);
+  }
   // Write configuration to an xconfig file in a standard location.
   await updateGeneratedXcodeProperties(
     project: flutterProject,
     buildInfo: buildInfo,
     useMacOSConfig: true,
+    symrootOverride: fs.path.join(flutterBuildDir.absolute.path, 'Build', 'Products'),
```

Comment:
Since the point of having this set in the xcconfig at all is presumably to ensure that Xcode UI builds go to the same place as `flutter` builds, this value should be extracted to a variable that's used both here and on line 46 to ensure they stay in sync.